### PR TITLE
Feature/us7414 map view search optimization

### PR DESCRIPTION
**Pull request by eymiha**
Figure out the right zoom over a location (the highest level at least 10 pins), and preserve zoom when location changes and the new search-again-here button is clicked.

### CODE REVIEW

File: src/app/components/map/map.component.ts [hunk 1]
```diff
@@ -34,19 +34,33 @@ export class MapComponent implements OnInit {
...
-          this.mapSettings.zoom = 15;
-          this.mapSettings.lat = pos.lat;
-          this.mapSettings.lng = pos.lng;
-        }
-      );
-    } else {
-      this.mapSettings.zoom = 15;
-      this.mapSettings.lat = this.searchResults.centerLocation.lat;
-      this.mapSettings.lng = this.searchResults.centerLocation.lng;
+    ////
```
Comment (plachmann) [inline src/app/components/map/map.component.ts]: Remove commented code

File: src/app/components/search-local/search-local.component.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      if ((update.value === "dragend") || (update.value === "zoom_changed")) {
+        this.mapView = update;
+        this.showButton();
+      }
+    });
+  }
+
+  public active: boolean;
+
+  ngOnInit() {
```
Comment (plachmann) [inline src/app/components/search-local/search-local.component.ts]: We need some unit tests for this new component

File: src/app/services/state.service.ts [hunk 3]
```diff
@@ -26,4 +27,12 @@ export class StateService {
   public getShowingPinCount() {
     return this.showingPinCount;
   }
+
```
Comment (plachmann) [inline src/app/services/state.service.ts]: add some unit tests for these new methods

File: src/app/components/map/map.component.ts (hunk 1 above)
Comment (eymiha) [inline src/app/components/map/map.component.ts]: commented code block removed.

File: src/app/components/search-local/search-local.component.ts (hunk 2 above)
Comment (eymiha) [inline src/app/components/search-local/search-local.component.ts]: tests added

File: src/app/services/state.service.ts (hunk 3 above)
Comment (eymiha) [inline src/app/services/state.service.ts]: tests added

**Pull request merged**